client/src/modules/cohort/index.jsx

```diff
@@ -81,17 +90,32 @@ class Cohort extends PureComponent {
       }
     } = this.props;
 
+    const fetchPromises = [];
+
     this.setState({ pendingForDetails: true });
 
-    return fetchCohortDetails(id)
+    const abortableFetchDetails = makeAbortablePromise(fetchCohortDetails(id));
+
+    this.pendingPromises.push(abortableFetchDetails);
+    fetchPromises.push(abortableFetchDetails.promise);
+
+    if (!this.checkIfArchived()) {
+      const abortableFetchLists = makeAbortablePromise(fetchListsMetaData(id));
+
+      this.pendingPromises.push(abortableFetchLists);
+      fetchPromises.push(abortableFetchLists.promise);
+    }
+
+    Promise.all(fetchPromises)
       .then(() => {
-        if (!this.checkIfArchived()) {
-          return fetchListsMetaData(id);
-        }
-      })
-      .finally(() => {
         this.handleBreadcrumbs();
         this.setState({ pendingForDetails: false });
+      })
+      .catch(err => {
+        if (!(err instanceof AbortPromiseException)) {
+          this.setState({ pendingForDetails: false });
+        }
+        noOp();
```

Comment:
@marekrozmus  is this noOp needed here? 
if an err occurs we just ignore them. Do we have to run noOp function here which takes time?

Reply:
I removed noOp